Monika-After-Story/MonikaModDev
# Gifted Clothes Selector

This unlocks the clothes selector for gifted clothes at Happy+ ~~so that you can ask her to wear clothes you gave her as a gift, while our special event clothes still stay locked until Love.~~

Also adds blazerless outfit 

Also fixes a potential crash on load if you were wearing a json outfit and deleted it. Now we just put on def and there is no crash.

We also now only change Monika's outfit for the `greeting_hairdown` if wearing a baked outfit.

## Testing
(recommend downloading the sprite pack from #4427 for testing)

- start a fresh persistent and save after the intro to save some time

- gift the formal dress on normal affection and confirm the selector is not available

- increase affection to Happy+ and confirm the selector is available and only includes the school uniform, blazerless, and any gifted outfits, and not sundress, santa, or either o31 outfits

- decrease affection below Happy and confirm the selector is not available and an event is pushed that changes back to school uniform. also confirm this happens if you start a session at Normal or lower while wearing anything except school uniform

- go back to the persistent before you gifted the dress, increase affection to Happy+ and give the formal dress

- verify that the selector unlocks immediately

- on a copy of your current persistent that has special outfits unlocked and at Love affection, gift the formal dress if you haven't already

- verify only one selector is available and all unlocked outfits are present

- decrease affection below Love and verify selector is present and only gifted outfits (and def/blazerless) are present

- wearing the formal dress, quit the game, delete the json, load the game and verify no crash and Moni is wearing her school uniform

- ~~verify if you drop below Love but not below Happy with no gifted outfits unlocked and are not wearing school uniform that the selector is available and that only school uniform is an option, and that after changing to school uniform, the selector locks~~
